fsw/cfe-core/src/sb/cfe_sb_api.c

```diff
@@ -540,6 +536,169 @@ int32 CFE_SB_GetPipeOpts(CFE_SB_PipeId_t PipeId, uint8 *OptsPtr)
     return CFE_SUCCESS;
 }/* end CFE_SB_GetPipeOpts */
 
+/******************************************************************************
+**  Function:  CFE_SB_GetPipeName()
+**
+**  Purpose:
+**    SB internal function to get the string of the pipe name of the given
+**    Pipe ID.
+**
+**  Arguments:
+**    MsgId       : ID of the message
+**    PipeNameBuf : Buffer to receive name, must be OS_MAX_API_NAME bytes long
+**
+**  Return:
+**    Will return CFE_SUCCESS and populate PipeNameBuf with the name
+**    of the pipe on success, otherwise PipeNameBuf is filled with NULLs
+**    and this returns CFE_SB_BAD_ARGUMENT on error.
+**
+*/
+int32 CFE_SB_GetPipeName(char *PipeNameBuf, size_t PipeNameSize, CFE_SB_PipeId_t PipeId){
+    OS_queue_prop_t queue_prop;
+    int32 Status = CFE_SUCCESS;
+    uint32 TskId = 0;
+    char FullName[(OS_MAX_API_NAME * 2)];
+
+    if(PipeNameBuf == NULL) {
+        CFE_EVS_SendEventWithAppID(CFE_SB_GETPIPENAME_NULL_PTR_EID, CFE_EVS_EventType_ERROR,
+            CFE_SB.AppId, "Pipe Name Error:NullPtr,Requestor %s",
+            CFE_SB_GetAppTskName(TskId,FullName));
+
+        Status = CFE_SB_BAD_ARGUMENT;
+    } else if(PipeId >= CFE_PLATFORM_SB_MAX_PIPES){
+        CFE_EVS_SendEventWithAppID(CFE_SB_GETPIPENAME_ID_ERR_EID, CFE_EVS_EventType_ERROR,
+            CFE_SB.AppId, "Pipe Id Error:Bad Argument,Id=%d,Requestor %s",
+            PipeId,CFE_SB_GetAppTskName(TskId,FullName));
+
+        memset(PipeNameBuf, 0, PipeNameSize);
+
+        Status = CFE_SB_BAD_ARGUMENT;
+    }else{
+        if (OS_QueueGetInfo(CFE_SB.PipeTbl[PipeId].SysQueueId, &queue_prop)
+                == OS_SUCCESS){
+            strncpy(PipeNameBuf, queue_prop.name, OS_MAX_API_NAME-1);
```

Comment:
This needs to use `PipeNameSize`, not OS_MAX_API_NAME-1.  It should also ensure a null-terminated output.

Reply:
fixed

Reply:
Hate to be a pain, but still need to make sure that the output is null terminated.  This should also handle the case if the passed in size is 0 (since it is handling a passed-in buffer of NULL, should handle this too).